### PR TITLE
Using Javascript to parse boolean

### PR DESCRIPTION
**Pull request by alex-deaconu**
Default value for boolean was not being applied if property is undefined
Using Javascript to properly test for property presence

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@olegrise my previous fix for this failed. I didn't take into account the use case where the property was missing. It would return `false`, rather than the default value. 

Updated to use Javascript to check the property. Checking `"true"`, `"false"` strings, checking if the property is there, and finally checking the boolean value. This should be more reliable.

Please review. Thanks.

File: src/com/risevision/common/client/json/JSOModel.java [hunk 1]
```diff
@@ -197,23 +197,25 @@ public final boolean getBoolean(String key) {
...
+	public final native boolean getBooleanNative(String key, Boolean defaultValue) /*-{
+		if (this[key] === "true") {
+			return true;
+		}
+		else if (this[key] === "false") {
+			return false;
+		}
+		else if (this.hasOwnProperty(key)) {
+			return this[key];
+		}
```
Comment (olegrise) [inline src/com/risevision/common/client/json/JSOModel.java]: @alex-deaconu what happens if `this[key]` value is not boolean? Will it automatically get converted to boolean?

Comment (alex-deaconu) [inline src/com/risevision/common/client/json/JSOModel.java]: @olegrise yes. Regular Javascript rules apply. Any value will return `true`, `null` will return `false`. Do you think checking the type or for `null` is necessary? Thanks.

Comment (olegrise) [inline src/com/risevision/common/client/json/JSOModel.java]: @alex-deaconu it depends on how you want `null` to be treated. Usually if value is `null`, then it should return default value, but it's up to you.

**Comment by alex-deaconu** [comment]
@olegrise you're totally right. See updated commit - `null` value will go to default.

However, any other non-boolean value will default to `true`. 

Thanks.

**Comment by olegrise** [comment]
Nice :+1:

**Pull request merged**